test/Dictionary.jl

```diff
@@ -377,4 +377,17 @@
         @test dictcopy == Dictionary([3, 2, 1], ['a', 'b', 'c'])
         @test all(isassigned(dictcopy, i) for i in keys(dictcopy))
     end
+
+    @testset "merge" begin
+        d1 = Dictionary([1, 2, 3], [1, 2, 3])
+        d2 = Float32.(d1)
+        d3 = Float64.(d1) .+ 0.5
+        @test merge(d1, d2) isa Dictionary{Int, Float32}
+        @test merge(d1, d2, d3) isa Dictionary{Int, Float64}
+        @test merge!(d2, d1) == d1
+        @test_throws InexactError merge!(d1, d2, d3)
+        @test mergewith(+, d1, d2, d3) isa Dictionary{Int, Float64}
```

Comment:
We might either want to bump compatiblity to Julia 1.6, or put a `isdefined(Base, :mergewith)` guard here?

Reply:
For the moment I'll just add `isdefined(Base, :mergewith)` here. Upping compatibility could be left for a different, standalone PR, since it is an important decision.